control_protocol.md

```diff
@@ -330,17 +330,490 @@ A Component MUST also offer a list of all possibly callable methods in accordanc
+                        ]
+                    },
+                    "required": false
+                }
+            ],
+            "result": {
+                "name": "result",
+                "schema": {
+                    "type": "null"
+                },
+                "required": true
+            }
+        },
+        {
+            "name": "stop_polling",
+            "params": [],
+            "result": {
+                "name": "result",
+                "schema": {
+                    "type": "null"
+                },
+                "required": true
+            }
+        },
+        {
+            "name": "get_polling_interval",
+            "params": [],
+            "result": {
+                "name": "result",
+                "schema": {
+                    "type": "number"
+                },
+                "required": true
+            }
+        },
+        {
+            "name": "set_polling_interval",
+            "params": [
+                {
+                    "name": "polling_interval",
+                    "schema": {
+                        "type": "number"
+                    },
+                    "required": true
+                }
+            ],
+            "result": {
+                "name": "result",
+                "schema": {
+                    "type": "null"
+                },
+                "required": true
+            }
+        },
+    ],
+    "components": {
+        "schemas": {}
+    }
+}
+:::
+
 
-### Messages for Transport Layer
+#### Locking Actor
 
-- SIGNIN
-- SIGNOUT
-- ACKNOWLEDGE
-- ERROR
-- PING
-- CO_SIGNIN
-- CO_SIGNOUT
+An {ref}`control_protocol.md#Actor` which support locking resources MUST offer the following methods.
 
 :::{note}
-TODO How to make these messages work? Define them directly in the transport layer?
+TBD change the python code to json
 :::
+
+::: python
+    def lock(self, resource: Optional[str] = None) -> bool: ...
+
+    def unlock(self, resource: Optional[str] = None) -> None: ...
+
+    def force_unlock(self, resource: Optional[str] = None) -> None: ...
+:::
+
+
+### Errors
+
+Every error has a code and a message.
+Additionally they may have a ``data`` field with more information.
+
+According to JSONRPC, applications can define error codes between -32000 and -32099.
+LECO defines the following errors.
+
+
+#### Routing errors
+
+Errors related to routing (mainly emitted by Coordinators).
+Their error codes are in the range of -32090 to -32099.
+
+
+| code   | message                            | data                 | description                                                                          |
+|--------|------------------------------------|----------------------|--------------------------------------------------------------------------------------|
+| -32090 | You did not sign in!               | -                    | If a Component did not sign in.                                                      |
```

Comment:
```suggestion
| -32090 | Component not signed in yet!       | Name of the component| If a Component did not sign in.                                                      |
```
?

Reply:
A Coordinator sends that error to a Component, if the Component is not signed in, such that the message makes sense from the Component's point of view.
And a Component does not need to be informed about its own name.

That was the reasoning behind this error message.

Reply:
Consider though that error messages can end up without that context, e.g. in an error log or bug report, so a little added context can be quite useful I think.

Reply:
We have the message header with the receiver name, however, that might get lost.

Ok, I'll change it.